File: c#/EmployeeManagement/Models/PTInstructor.cs
namespace EmployeeManagement.Models
{
    // Part-time instructor employee with a maximum weekly hour limit.
    public class PTInstructor : PTEmployee
    {
        public int MaxHours { get; }

        public PTInstructor(int id, string firstName, string lastName, DateOnly dateOfBirth, string position, Departments department, double rate, int hours, int maxHours)
            : base(id, firstName, lastName, dateOfBirth, position, department, rate, hours)
        {
            if (maxHours < 0)
            {
                throw new System.ArgumentException("Max hours cannot be negative.", nameof(maxHours));
            }

            MaxHours = maxHours;
        }

        public override void ReportToManager()
        {
            Console.WriteLine($"PT Instructor {FirstName} {LastName} reports to the manager with a maximum of {MaxHours} weekly hours.");
        }

        public override string ToString()
        {
            return base.ToString() + $", MaxHours={MaxHours}";
        }
    }
}

File: Java/PTInstructor.java
public class PTInstructor extends PTEmployee {

    private int max_hours;

    public PTInstructor(
        int id,
        String first_name,
        String last_name,
        String dob,
        String position,
        Departments department,
        double rate,
        int hours,
        int max_hours
    ) {
        super(id, first_name, last_name, dob, position, department, rate, hours);
        this.max_hours = max_hours;
    }

    @Override
    public void reportToManager() {
    }
}
